R/graded.R

```diff
@@ -9,54 +9,54 @@
 graded <- function(correct, message = NULL) {
   chkm8_single_character(message)
   checkmate::expect_logical(correct, any.missing = FALSE, len = 1, null.ok = FALSE)
-  
-  structure(
-    list(message = message %||% "", correct = correct),
-    class = "grader_graded"
+
+  obj <- structure(
+    list(
+      message = message %||% "",
+      correct = correct
+    ),
+    class = c("gradethis_graded", "condition")
   )
+
+  # _throw_ condition object
+  # also pretty prints the condition
+  signalCondition(obj)
+
+  # return the object
+  obj
 }
 
-is_grade <- function(x) {
-  inherits(x, "grader_graded")
+is_graded <- function(x) {
+  inherits(x, "gradethis_graded")
 }
 
 
-#' Provide grade feedback
-#'
-#' Creates a feedback object suitable for returning in a learnr checking function
-#' (e.g., the `exercise.checker` option in [learnr::tutorial_options()])
-#'
-#' @param grade a [graded()] object.
-#' @param type Feedback type (visual presentation style). Can be "auto", "success", "info", "warning", "error", or "custom". 
-#' Note that "custom" implies that the "message" field is custom HTML rather than a character vector.
-#' @param location Location for feedback ("append", "prepend", or "replace").
+#' @rdname grade_result
 #' @export
-grade_feedback <- function(grade,
-                           type = c("auto", "success", "info", "warning", "error", "custom"), 
-                           location = c("append", "prepend", "replace")) {
-  
-  if (!is_grade(grade)) {
-    stop("`grade` must be a `graded` object", call. = FALSE)
-  }
-  
-  type <- match.arg(type)
-  
-  if (identical("auto", type)) {
-    type <- if (grade$correct) "success" else "error"
-  }
-  
-  structure(
-    list(
-      message = grade$message, 
-      correct = grade$correct, 
-      type = type, 
-      location = match.arg(location)
-    ),
-    class = "grader_feedback"
-  )
+pass <- function(message = getOption("gradethis.pass", "Correct!")) {
+  graded(message = glue_with_env(parent.frame(1), message), correct = TRUE)
 }
+#' @rdname grade_result
+#' @export
+fail <- function(message = getOption("gradethis.fail", "Incorrect")) {
+  graded(message = glue_with_env(parent.frame(1), message), correct = FALSE)
+}
```

Comment:
Just in case people want to do crazy metaprogramming

```suggestion
fail <- function(message = getOption("gradethis.fail", "Incorrect"), env = parent.frame()) {
  graded(message = glue_with_env(env, message), correct = FALSE)
}
```

Reply:
Do similar for `pass()`